src/store/flow/block.ts

```diff
@@ -122,9 +130,32 @@ export const mutations: MutationTree<IFlowsState> = {
 }
 
 export const actions: ActionTree<IFlowsState, IRootState> = {
-  block_setLabel({commit}, {blockId, value}) {
+  block_setLabel({commit, dispatch}, {blockId, value}) {
     commit('block_setLabel', {blockId, value})
-    commit('block_setName', {blockId, value: snakeCase(value)})
+    dispatch('block_setName', {blockId, value: snakeCase(value)})
+  },
+
+  block_setName(
+    {commit, state},
+    {blockId, value, isUserDefinedName = false}: {blockId: IBlock['uuid'], value: string, isUserDefinedName: boolean},
+  ) {
+    const block = findBlockOnActiveFlowWith(blockId, state as unknown as IContext)
+    const canSetName = block.ui_metadata?.has_user_defined_name !== true
+
+    if (canSetName || isUserDefinedName) {
```

Comment:
@object-required , If I'm not wrong, the logic here is `if the user hasn't defined it yet, then auto update the name from the payload value`, right ?

With that, we know that the **final state** we want to know is "should we auto update the name?". What if we use that final state to rename the flag by using:
- `should_auto_update_name` instead of `has_user_defined_name`
- `block_autoUpdateName`  instead of `block_setUserDefinedName`
- `shouldAutoUpdateName` instead of `canSetName` (which is more like permission name). Or we can even avoid creating a variable for that, it's only used once.

That would be easier to read I think, your thoughts?

- [ ] rename the flag if you agree